analyzers/src/SonarAnalyzer.Common/Common/AnalyzerConfiguration.cs

```diff
@@ -76,38 +81,35 @@ public bool IsEnabled(string ruleKey)
                     && enabledRules.Contains(ruleKey);
             }
 
-            public void Initialize(AnalyzerOptions options)
+            public void Initialize(AnalyzerOptions options, IRuleLoader ruleLoader)
             {
-                if (isInitialized)
+                var currentSonarLintXmlPath = GetSonarLintXmlPath(options);
+                if (isInitialized && loadedSonarLintXmlPath == currentSonarLintXmlPath)
                 {
                     return;
                 }
 
                 lock (@lock)
```

Comment:
Can we rename the lock? I do not like @keyword identifier names... and besides that it validates S1311 that says static readonly fields should begin with upper-case letter.